src/init.cpp

```diff
@@ -723,6 +727,10 @@ static void ThreadImport(std::vector<fs::path> vImportFiles)
         if (file) {
             LogPrintf("Importing blocks file %s...\n", path.string());
             LoadExternalBlockFile(chainparams, file);
+            if (ShutdownRequested()) {
+                LogPrintf("Shutdown requested. Exit %s\n", __func__);
+                return;
+            }
```

Comment:
Shouldn't this block be outside of the `if ... else ...` statement?

Reply:
why? That would be a slight behaviour change. Previously, we'd also loop over invalid files until the first valid file is found and then check for interruption. I don't think it matters much in any case.